src/main/java/appeng/block/AEBaseSlabBlock.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.block;
+
+
+import java.util.EnumSet;
+import java.util.Random;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockSlab;
+import net.minecraft.client.renderer.texture.IIconRegister;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.world.World;
+import appeng.core.features.AEFeature;
+import appeng.core.features.IAEFeature;
+import appeng.core.features.IFeatureHandler;
+import appeng.core.features.SlabBlockFeatureHandler;
+
+
+public class AEBaseSlabBlock extends BlockSlab implements IAEFeature
+{
+	private final IFeatureHandler features;
+	public AEBaseBlock block;
+	public int meta;
+	public boolean isDoubleSlab;
+	public AEBaseSlabBlock slabs;
+	public AEBaseSlabBlock dSlabs;
+	public final String name;
+
+	public AEBaseSlabBlock( AEBaseBlock block, int meta, EnumSet<AEFeature> features, boolean isDoubleSlab, String name )
+	{
+        super( isDoubleSlab, block.getMaterial() );
+		this.block = block;
+		this.meta = meta;
+		this.name = name;
+		this.isDoubleSlab = isDoubleSlab;
+		this.setBlockName( "appliedenergistics2." + name );
+        this.setHardness( block.getBlockHardness( null, 0, 0, 0 ) );
```

Comment:
Why call the getters, if you do not change anything? Same for the `setResistance`.